### ISSUE TITLE
Sharing statement when using method chains should be a configuration option

### ISSUE THREAD

**Issue by fbouton** [go-gorm/gorm#3769]
## Describe the feature

V2 Changes include "To reduce GC allocs, GORM V2 will share Statement when using method chains, and will only create new Statement instances for new initialized *gorm.DB or after a New Session Method, to reuse a *gorm.DB, you need to make sure it just after a New Session Method"

We propose that a global configuration allows programmers to maintain previous behavior. Something like : 

`&gorm.Config{ LegacyMethodChaining: true}`

## Motivation

Basically, now we have to use a lot of `DB = DB.Session(&gorm.Session{NewDB: true})` in our methods. Our software is already in production and it is quite big. Upgrading to Gorm V2 would have too many side effects. So, being able to maintain the previous behavior would allow us to easier upgrade.

## Related Issues

https://gorm.io/docs/v2_release_note.html

**Comment by jinzhu** [comment]
Hi,

Could you make some examples where to write those `DB = DB.Session(&gorm.Session{NewDB: true})`

In most cases, I think it is not necessary to do that.

**Comment by jinzhu** [comment]
And this would be confusion when using `*gorm.DB` in `Hooks`, `callbacks`, so not going to accept the request.

Thank you.

**Comment by fbouton** [comment]
> And this would be confusion when using `*gorm.DB` in `Hooks`, `callbacks`, so not going to accept the request.
> 
> Thank you.

Well, at least thanks for your reply. We are still working on the migration, thank you for your work ;-)
